docs/blog-1/blog-1.md

```diff
@@ -0,0 +1,253 @@
+
+TL;DR: If you work a with a git rebase flow, `git machete` will (vastly!) help you manage the jungle of branches stacking on top of each other when you're, for example, working on a couple of different PRs in parallel.
+
+
+## The problem
+
+In day-to-day development it is common that for various reasons coders have to work with multiple git branches at the same time.
+I'll show a repository with a bunch of topic branches so as to demonstrate the typical problems that can occur under such conditions
+and how a few nifty cuts of `git machete` will prevent us from getting into a complete _git jungle_.
+The example I'm about to provide is deliberately exaggerated - in fact, most likely you'll have at most 3 or 4 branches developed simultaneously.
+Still, `git machete` is significantly helpful, even in less complex scenarios.
+
+Let's assume you have implemented a fix on a branch `adjust-reads-prec` (the actual content of the fix is completely irrelevant) and made a pull request to `develop`.
+While the PR was under review, you started work on another topic branch `block-cancel-order`.
+
+Unfortunately, the changes that you intended to introduce on `block-cancel-order` depended on what had already been done on `adjust-reads-prec`...
+So you forked a new branch (`block-cancel-order`) off `adjust-reads-prec` and, when the change was ready, you made another PR, this time for `block-cancel-order` to `adjust-reads-prec`.
+
+In the meantime, reviewers posted their comments on the first PR.
+You applied their remarks as a `1st round of fixes` commit on the `adjust-reads-prec` branch.
+
+Let's see how the state of the repo looks in a graphical tool like `gitk`:
+
+![gitk](gitk-0.png)
+
+In most cases it's rather unusual to have more than 2 branches stacked on top of each other, but for the sake of completeness let's now make this chain longer.
+You developed a couple of new refactors and bug fixes - say, on branches `change-table` and `drop-location-type`.
+As each of them was dependent on changes already waiting in the review queue, you began stacking another branches on top of each other.
+You ended up having 4 branches in the chain, each dependent on the previous one: `adjust-reads-prec`, `block-cancel-order`, `change-table` and `drop-location-type`.
+
+Now for the sake of providing a full-fledged example, let's also add a couple of other branches (not related to the 4-branch chain) to our scenario.
+Let's assume you also independently developed a feature `edit-margin-not-allowed` and later derived a branch `full-load-gatling` from that point.
+Also, you created a branch `grep-errors-script` that (fortunately!) nothing depended on, and a `hotfix/remove-trigger` branch, but on top of `master`, not `develop`.
+
+If you'd like to quickly recreate this repository state on your local machine, you can use [this handy script (link)](https://raw.githubusercontent.com/VirtusLab/git-machete/copy-docs/docs/blog-1/sandbox-setup.sh),
```

Comment:
Again, all links pointing to `copy-docs` branch should be updated (with a link to `develop` branch) on a subsequent PR once this PR is merged